cli/module_registry.go

```diff
@@ -425,6 +444,12 @@ func validateModuleFile(client *viamClient, moduleID moduleID, tarballPath, vers
 		// if path == entrypoint, we have found the right file
 		if filepath.Clean(path) == filepath.Clean(entrypoint) {
 			info := header.FileInfo()
+			if info.IsDir() {
+				return errors.Errorf(
+					"the module archive contains a directory at the entrypoint %q instead of an executable file",
+					entrypoint)
+			}
+
```

Comment:
Flyby: I figured out how to replicate Alexa's issue if you manually create the tarball instead of the auto-bundling.

 ```
Error validating module: the module archive contains a directory at the entrypoint "testdir" instead of an executable file. For more details, please visit: https://docs.viam.com/fleet/cli/#module
```

I can break this out into another PR if you want.